packages/next/src/server/response-cache/utils.ts

```diff
@@ -62,7 +64,14 @@ export async function toResponseCacheEntry(
               html: RenderResult.fromStatic(response.value.html),
               rscData: response.value.rscData,
               headers: response.value.headers,
-              status: response.value.status,
+              status:
+                // RSC redirect is embedded in the payload itself and handled by client router directly
+                // This makes it consistent with how it's handled in minimalMode
+                isRscRequest &&
+                response.value.status &&
+                RedirectStatusCode[response.value.status]
+                  ? 200
+                  : response.value.status,
```

Comment:
we might want to move this status code massaging higher up so it's done after we get the cache value and are sending the response to avoid mutating the cache entry